docs/okeps/okep-50088-evpn.md

```diff
@@ -0,0 +1,422 @@
+ip link set br101 addr aa:bb:cc:00:00:64
+ip link add vni101 type vxlan local 100.64.0.1 dstport 4789 id 101 nolearning
+ip link set vni101 master br100 addrgenmode none
+ip link set vni101 type bridge_slave neigh_suppress on learning off
+ip link set vni101 up
+ip link set br101 up
+```
+
+Note, it is not required to wire the OVN logical switch to the linux bridge in this case. It is also not required to
+modify routes in ovn_cluster_router. Pod egress traffic should be rerouted towards mpx as is done today with BGP.
+
+The FRR configuration remains the same as the previous example.
+
+An IP-VRF with a layer 3 UDN would look something like this:
+
+![](images/l3evpn.png)
+
+In this case each node has its own layer 2 domain, and routing is used via the IP-VRF for inter-node UDN communication. 
+
+#### MAC-VRF + Layer 2 UDN
+
+With only a MAC-VRF the node configuration changes somewhat:
+
+```bash
+## Create the MAC-VRF
+ip link add br100 type bridge
+ip link set br100 addrgenmode none
+ip link add vni100 type vxlan local 100.64.0.1 dstport 4789 id 100 nolearning
+ip link set vni100 master br100 addrgenmode none
+ip link set vni100 type bridge_slave neigh_suppress on learning off
+sysctl -w net.ipv4.conf.br100.forwarding=0
+sysctl -w net.ipv6.conf.br100.forwarding=0
+ip link set vni100 up
+ip link set br100 up
+
+## Connect OVS to the linux bridge
+ip link add blue-0 type veth peer name blue-1
+ovs-vsctl add-port br-int blue-0
+ip link set dev blue-1 master br100
+ip link set dev br100 up
+ip link set dev blue-1 up
+```
+
+In the above config, the bridge is no longer enslaved to an IP-VRF. Additionally, there is no address assigned to the bridge
+SVI interface and IP forwarding has been disabled on the bridge.
+
+The same OVS connection to linux bridge is still required, and ovnkube-controller will still need to create the OVN logical
+switch port attached to the worker. However, ovnkube-controller will not be required to modify routes in ovn_cluster_router.
+
+The FRR configuration will not need to specify the "vrf" stanza for this MAC-VRF to function.
+
+Architecturally, the traffic pattern and topology will look the same as the diagram in the
+[MAC-VRF + IP-VRF Combination with Layer 2 UDN](#l2evpn-anchor) section.
+
+### Testing Details
+
+The EVPN feature will require E2E tests to be written which will simulate a spine and leaf topology that the KIND Kubernetes
+nodes are attached to. From there tests will be added that will create UDN+BGP+EVPN and test the following:
+
+1. UDN pods are able to talk to external applications on the same VPN.
+2. UDN pods are unable to talk to external applications on a different VPN.
+3. UDN pods are able to talk to other UDN pods on the same network without a Geneve overlay via EVPN.
+4. The above tests will apply for both IP-VRF and MAC-VRF EVPN types.
+5. For MAC-VRF, a test will be added to ensure VM migration between two Kubernetes nodes with EVPN.
+6. Testing with EgressIP.
```

Comment:
Is this something EgressIP on UDN can be advertised by EVPN, just like EgressIP on BGP?  If so, do we have specific configuration for it or just using BGP RA advertisement EgressIP?